src/main/scala/strawman/collection/immutable/ImmutableArray.scala

```diff
@@ -0,0 +1,102 @@
+package strawman.collection.immutable
+
+import strawman.collection.mutable.ArrayBuffer
+import strawman.collection.{IterableFactory, IterableOnce, Iterator, View}
+
+import scala.{Any, Boolean, Int, Nothing}
+import scala.runtime.ScalaRunTime
+import scala.Predef.{???, intWrapper}
+
+/**
+  * An immutable array.
+  *
+  * Supports efficient indexed access and has a small memory footprint.
+  */
+class ImmutableArray[+A] private (private val elements: scala.Array[Any]) extends IndexedSeq[A] with SeqOps[A, ImmutableArray, ImmutableArray[A]] {
+
+  protected[this] def fromIterable[E](it: strawman.collection.Iterable[E]): ImmutableArray[E] = ImmutableArray.fromIterable(it)
+
+  protected[this] def fromSpecificIterable(coll: strawman.collection.Iterable[A]): ImmutableArray[A] = ImmutableArray.fromIterable(coll)
+
+  def length: Int = elements.length
+
+  override def knownSize: Int = elements.length
+
+  def apply(i: Int): A = ScalaRunTime.array_apply(elements, i).asInstanceOf[A]
+
+  def iterator(): Iterator[A] = view.iterator()
+
+  override def map[B](f: A => B): ImmutableArray[B] = ImmutableArray.tabulate(length)(i => f(apply(i)))
+
+  override def flatMap[B](f: A => IterableOnce[B]): ImmutableArray[B] = ImmutableArray.fromIterable(View.FlatMap(coll, f))
+
+  def :+ [B >: A](elem: B): ImmutableArray[B] = {
+    val dest = scala.Array.ofDim[Any](length + 1)
+    java.lang.System.arraycopy(elements, 0, dest, 0, length)
+    dest(length) = elem.asInstanceOf[Any]
+    new ImmutableArray(dest)
+  }
```

Comment:
That's precisely the use case Spandex (#52) is designed for. It also has a method that wraps an existing array without copying (its package private since it is unsafe if the array is later modified from the outside) so it should support usage for varargs. The performance is only unpredictable in so far as an append or prepend operation will result in an array copy only when the array is full on the affected side or if the targeted array slot is already occupied (by a non eq value/reference).